Flow.Launcher/ViewModel/SettingWindowViewModel.cs

```diff
@@ -125,25 +125,44 @@ public void Save()
         #region general
 
         // todo a better name?
-        public class LastQueryMode
+        public class LastQueryMode : BaseModel
         {
             public string Display { get; set; }
             public Infrastructure.UserSettings.LastQueryMode Value { get; set; }
         }
+
+        private List<LastQueryMode> _lastQueryModes = new List<LastQueryMode>();
         public List<LastQueryMode> LastQueryModes
         {
             get
             {
-                List<LastQueryMode> modes = new List<LastQueryMode>();
-                var enums = (Infrastructure.UserSettings.LastQueryMode[])Enum.GetValues(typeof(Infrastructure.UserSettings.LastQueryMode));
-                foreach (var e in enums)
+                if (_lastQueryModes.Count == 0)
                 {
-                    var key = $"LastQuery{e}";
-                    var display = _translater.GetTranslation(key);
-                    var m = new LastQueryMode { Display = display, Value = e, };
-                    modes.Add(m);
+                    _lastQueryModes = InitLastQueryModes();
                 }
-                return modes;
+                return _lastQueryModes;
```

Comment:
Thanks for the fix!
Though, I didn't see why this can fix the issue. It seems like you only add a cache for that (which is super reasonable though)?
Would you please explain a bit?

Reply:
1.LastQueryMode add INotifyPropertyChanged interface
public class LastQueryMode : BaseModel

2.when modifying the language,call UpdateLastQueryModeDisplay();
        private void UpdateLastQueryModeDisplay()
        {
            foreach (var item in LastQueryModes)
            {
                item.Display = _translater.GetTranslation($"LastQuery{item.Value}");
            }
        }

        public string Language
        {
...
            set
            {
 ...
                UpdateLastQueryModeDisplay();//when modifying the language,
            }
        }